### PR TITLE
Enable deletion of individual questions inside of form

### PR DESCRIPTION
**Pull request by sik-leanix**
As a user I would like to be able to remove a question after pressing “add question” as I might’ve done that by accident and then have no possibility to remove it again.

image

### CODE REVIEW

File: Day-11-Quiz-App/quizBuilder.js [hunk 1]
```diff
@@ -122,8 +123,10 @@ class QuizBuilder {
     }
 
     _addInputElementsForOneQuestion(questionData) {
-        this._questionsContainer = document.getElementById("questionsContainer");
 
+        this._questionsContainer = document.getElementById("questionsContainer");
+        
+        const createHR = document.createElement("hr");
```
Comment (konstantintieber) [inline Day-11-Quiz-App/quizBuilder.js:129]: variable naming could be improved here. The fact that it's created is onlys valid during until we call `document.createElement`. Afterwards it isn't.

I think it's better to call it by its function: `separatorHr`

File: Day-11-Quiz-App/quizBuilder.js [hunk 2]
```diff
@@ -145,7 +144,37 @@ class QuizBuilder {
...
         this._questionsContainer.appendChild(questionTextHeader);
+        
+        // Here starts the delete Button
+        if (this._pressedAddQuestionButtonCounter != 0) {
+            const removeQuestionButton = document.createElement("button");
+            this._questionsContainer.appendChild(removeQuestionButton);
+            removeQuestionButton.textContent = "X";
+            removeQuestionButton.className = "removeQuestionButton";
+            choicesHeader.style.display = "inline-block";
+                removeQuestionButton.addEventListener("click", (event) => {
```
Comment (konstantintieber) [inline Day-11-Quiz-App/quizBuilder.js:157]: The `addEventListener` statement is intended one level to far in this case.
Only the function body should be indented.

File: Day-11-Quiz-App/quizBuilder.js [hunk 3]
```diff
@@ -145,7 +144,37 @@ class QuizBuilder {
...
+                event.preventDefault();
+                questionTextHeader.remove();
+                questionText.remove();
+                choicesHeader.remove();
+                choices.remove();
+                createHR.remove();
+                answerHeader.remove();
+                answer.remove();
+                error.remove()
+                removeQuestionButton.remove();error
```
Comment (konstantintieber) [inline Day-11-Quiz-App/quizBuilder.js:167]: Here you have another `error` statement at the end of the line that doesn't seem to be used.

File: Day-11-Quiz-App/quizBuilder.js [hunk 4]
```diff
@@ -168,7 +197,7 @@ class QuizBuilder {
         questionText.className = "inputStyles questionText";
         choices.className = "inputStyles questionChoices";
         answer.className = "inputStyles questionAnswer";
-        error.className = "error"
+        error.className = "error"
```
Comment (konstantintieber) [inline Day-11-Quiz-App/quizBuilder.js:200]: Here unneeded whitespace was added.

**Review by konstantintieber — CHANGES_REQUESTED** [review]

File: Day-11-Quiz-App/quizBuilder.js (hunk 4 above)
Comment (sik-leanix) [reply Day-11-Quiz-App/quizBuilder.js]: Done in https://github.com/sik-leanix/100-days-of-javascript/pull/5/commits/6e08ec485496313f8cb74cdba0ede9ddc3153baf

**Review by konstantintieber — APPROVED** [review]
Woohoo! Merge it

**Pull request merged**